devicetypes/smartthings/zigbee-valve.src/zigbee-valve.groovy

```diff
@@ -88,38 +90,57 @@ def parse(String description) {
         } else if (event.name == "powerSource") {
             event.value = event.value.toLowerCase()
         }
-
         sendEvent(event)
     }
     else {
         def descMap = zigbee.parseDescriptionAsMap(description)
         if (descMap.clusterInt == CLUSTER_BASIC && descMap.attrInt == BASIC_ATTR_POWER_SOURCE){
             def value = descMap.value
             if (value == "01" || value == "02") {
-                sendEvent(name: "powerSource", value: "mains")
+                sendEvent(name: "powerSource", value: "Mains")
             }
             else if (value == "03") {
-                sendEvent(name: "powerSource", value: "battery")
+                sendEvent(name: "powerSource", value: "Battery")
             }
             else if (value == "04") {
-                sendEvent(name: "powerSource", value: "dc")
+                sendEvent(name: "powerSource", value: "DC")
             }
             else {
-                sendEvent(name: "powerSource", value: "unknown")
+                sendEvent(name: "powerSource", value: "Unknown")
             }
         }
         else if (descMap.clusterInt == CLUSTER_POWER && descMap.attrInt == POWER_ATTR_BATTERY_PERCENTAGE_REMAINING) {
             event.name = "battery"
             event.value = Math.round(Integer.parseInt(descMap.value, 16) / 2)
             sendEvent(event)
         }
+        else if (descMap.clusterInt == zigbee.IAS_ZONE_CLUSTER && descMap.attrInt == zigbee.ATTRIBUTE_IAS_ZONE_STATUS) {
```

Comment:
I want to confirm that you have tested this yourself and that it works.

Reply:
Yes, i test it, we will receive this message